test/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopExe.cs

```diff
@@ -93,6 +93,87 @@ public class GivenThatWeWantToBuildADesktopExe : SdkTest
         }
 
         [Theory]
+
+        // implict rid with option to append rid to output path off -> do not append
+        [InlineData("implicitOff", "", false, false)]
+
+        // implicit rid with option to append rid to output path on -> do not append (never append implicit rid irrespective of option)
+        [InlineData("implicitOn", "", true, false)]
+
+        // explicit  rid with option to append rid to output path off -> do not append
+        [InlineData("explicitOff", "win7-x86", false, false)]
+        
+        // explicit rid with option to append rid to output path on -> append
+        [InlineData("explicitOn", "win7-x64", true, true)]
+        public void It_appends_rid_to_outdir_correctly(string identifier, string rid, bool useAppendOption, bool shouldAppend)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return;
+            }
+
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("DesktopMinusRid", identifier: Path.DirectorySeparatorChar + identifier)
+                .WithSource()
+                .WithProjectChanges(project =>
+                {
+                    var ns = project.Root.Name.Namespace;
+                    var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+                    propertyGroup.Add(new XElement(ns + "RuntimeIdentifier", rid));
+                    propertyGroup.Add(new XElement(ns + "AppendRuntimeIdentifierToOutputPath", useAppendOption.ToString()));
```

Comment:
Should we be testing the case where `AppendRuntimeIdentifierToOutputPath` is not specified in the project, as that will be the common case?